packages/contracts-bedrock/src/cannon/libraries/MIPS64Instructions.sol

```diff
@@ -213,376 +241,386 @@ library MIPS64Instructions {
     }
 
     /// @notice Execute an instruction.
-    function executeMipsInstruction(
-        uint32 _insn,
-        uint32 _opcode,
-        uint32 _fun,
-        uint64 _rs,
-        uint64 _rt,
-        uint64 _mem
-    )
-        internal
-        pure
-        returns (uint64 out_)
-    {
+    function executeMipsInstruction(ExecuteMipsInstructionParams memory _args) internal pure returns (uint64 out_) {
         unchecked {
-            if (_opcode == 0 || (_opcode >= 8 && _opcode < 0xF) || _opcode == 0x18 || _opcode == 0x19) {
+            if (
+                _args.opcode == 0 || (_args.opcode >= 8 && _args.opcode < 0xF) || _args.opcode == 0x18
+                    || _args.opcode == 0x19
+            ) {
+                uint32 opcode = _args.opcode;
```

Comment:
(nit) looks like we only use `opcode` in the switch statement, probably don't need an extra variable for it here?

Reply:
This is a stack copied from the `_args.opcode` memory value. While you can use mload/mstore operations in yul, you can't reference variables that reside in memory.